swift/apple/FirezoneKit/Sources/FirezoneKit/Helpers/IPCClient.swift

```diff
@@ -61,27 +69,146 @@ class IPCClient {
       options.merge(["token": token as NSObject]) { _, new in new }
     }
 
-    // Pass pre-1.4.0 Firezone ID if it exists. Pre 1.4.0 clients will have this
-    // persisted to the app side container URL.
-    if let id = FirezoneId.load(.pre140) {
-      options.merge(["id": id as NSObject]) { _, new in new }
-    }
-
     try session().startTunnel(options: options)
   }
 
-  func signOut() throws {
-    try session([.connected, .connecting, .reasserting]).stopTunnel()
-    try session().sendProviderMessage(encoder.encode(ProviderMessage.signOut))
+  func signOut() async throws {
+    try await withCheckedThrowingContinuation { continuation in
+      do {
+        try session().sendProviderMessage(encoder.encode(ProviderMessage.signOut)) { _ in
+          continuation.resume()
+        }
+      } catch {
+        Log.error(error)
+        continuation.resume(throwing: error)
+      }
+    }
   }
 
   func stop() throws {
     try session([.connected, .connecting, .reasserting]).stopTunnel()
   }
 
-  func toggleInternetResource(enabled: Bool) throws {
-    try session([.connected]).sendProviderMessage(
-      encoder.encode(ProviderMessage.internetResourceEnabled(enabled)))
+  func getConfiguration() async throws -> Configuration? {
+    let decoder = PropertyListDecoder()
+
+    return try await withCheckedThrowingContinuation { continuation in
+      do {
+        try session().sendProviderMessage(
+          encoder.encode(ProviderMessage.getConfiguration(configurationHash))
+        ) { data in
+          guard let data = data
+          else {
+            // Configuration hasn't changed
+            continuation.resume(returning: self.configurationCache)
+            return
+          }
+
+          // Compute new hash
+          self.configurationHash = Data(SHA256.hash(data: data))
+
+          do {
+            let decoded = try decoder.decode(Configuration.self, from: data)
+            self.configurationCache = decoded
+            continuation.resume(returning: decoded)
+          } catch {
+            continuation.resume(throwing: error)
+          }
+        }
+      } catch {
+        continuation.resume(throwing: error)
+      }
+    }
+  }
+
+  func setAuthURL(_ authURL: URL) async throws {
+    try await withCheckedThrowingContinuation { continuation in
+      do {
+        try session().sendProviderMessage(
+          encoder.encode(ProviderMessage.setAuthURL(authURL))
+        ) { _ in
+          continuation.resume()
+        }
+      } catch {
+        Log.error(error)
+        continuation.resume(throwing: error)
+      }
+    }
+  }
+
+  func setApiURL(_ apiURL: URL) async throws {
```

Comment:
Surely we can extract a helper function here that simply takes a `ProviderMessage`?

Reply:
Yeah, that should be a quick fix. I'll do that.

Reply:
Simplified to a `sendMessageWithoutResponse` private func.